crates/node/src/command.rs

```diff
@@ -210,59 +208,74 @@ pub fn run() -> sc_cli::Result<()> {
+                if let Ok(ref src_path) = local_path {
+                    let src_path = src_path.clone() + "/configs/genesis-assets/" + &asset.name;
+                    utils::copy_from_filesystem(
+                        src_path,
+                        madara_path.clone() + "/configs/genesis-assets",
+                        cli.run.update_configs,
+                    )?;
+                } else if !cli.run.disable_url_fetch {
+                    configs::fetch_and_validate_file(
+                        madara_configs.remote_base_path.clone(),
+                        asset,
+                        madara_path.clone() + "/configs/genesis-assets/",
+                        cli.run.update_configs,
                     )?;
                 }
             }
 
             if let (Some(chain_spec_url), None) = (cli.run.chain_spec_url.clone(), cli.run.testnet) {
-                utils::fetch_from_url(chain_spec_url, madara_path.clone() + "/chain-specs")?;
-            }
-
-            if let Some(Testnet::Sharingan) = cli.run.testnet {
-                let src_path = utils::get_project_path();
-                if let Ok(src_path) = src_path {
-                    let src_path = src_path + "/configs/chain-specs/testnet-sharingan-raw.json";
-                    utils::copy_from_filesystem(src_path, madara_path.clone() + "/chain-specs")?;
-                } else {
-                    utils::fetch_from_url(
-                        constants::SHARINGAN_CHAIN_SPEC_URL.to_string(),
+                utils::fetch_from_url(
+                    chain_spec_url.clone(),
+                    madara_path.clone() + "/chain-specs",
+                    cli.run.update_configs,
+                )?;
+                let chain_spec = chain_spec_url.split('/').last().expect("Chain spec file name not found");
+                cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/" + chain_spec);
+            } else if let (Some(Testnet::Sharingan), false) = (cli.run.testnet, cli.run.disable_url_fetch) {
+                if let Ok(ref src_path) = local_path {
+                    let src_path = src_path.clone() + "/configs/chain-specs/testnet-sharingan-raw.json";
+                    utils::copy_from_filesystem(
+                        src_path,
                         madara_path.clone() + "/chain-specs",
+                        cli.run.update_configs,
                     )?;
+                } else {
+                    // If testnet flag is specified, right now it's only Sharingan
+                    for chain_spec in madara_configs.chain_specs {
+                        configs::fetch_and_validate_file(
+                            madara_configs.remote_base_path.clone(),
+                            chain_spec,
+                            madara_path.clone() + "/configs/chain-specs/",
+                            cli.run.update_configs,
+                        )?;
+                    }
                 }
```

Comment:
This could be a match of `cli.run.testnet`, `cli.run.chain_spec_url` and `cli.run.disable_url_fetch`